src/js/views/maps/ViewfinderView.js

```diff
@@ -0,0 +1,154 @@
+     * of panning to different locations on the map. 
+     */
+    initialize(options) {
+      this.mapModel = options.model;
+    },
+
+    render() {
+      this.el.innerHTML = _.template(Template)(this.templateVars);
+
+      this.focusInput();
+    },
+
+    focusInput() {
+      const input = this.getInput();
+      input.focus();
+      // Move cursor to end of input.
+      input.val("");
+      input.val(this.templateVars.inputValue);
+    },
+
+    getInput() {
+      return this.$el.find(`.${classNames.input}`);
+    },
+
+    getButton() {
+      return this.$el.find(`.${classNames.button}`);
+    },
+
+    /** Event handler for Backbone.View configuration. */
+    keyup(event) {
+      if (event.key === "Enter") {
+        this.search();
+      }
+    },
+
+    valueChange() {
+      this.templateVars.inputValue = this.getInput().val();
+    },
+
+    /** Event handler for Backbone.View configuration. */
+    search() {
+      this.clearError();
+
+      const coords = this.parseValue(this.templateVars.inputValue)
+      if (!coords) return;
+
+      this.model.zoomTo({ ...coords, height: 10000 /* meters */ });
+    },
+
+    /**
+     * Parse the user's input as a pair of floating point numbers. Log errors to the UI
+     * @return {{Number,Number}|undefined} Undefined represents an irrecoverable user input,
+     *   otherwise returns a latitude, longitude pair.
+     */
+    parseValue(value) {
+      const matches = value.match(floatsRegex);
+      const hasBannedChars = value.match(bannedCharactersRegex) != null;
+      if (matches?.length !== 2 || isNaN(matches[0]) || isNaN(matches[1]) || hasBannedChars) {
+        this.setError("Try entering a search query with two numerical values representing a latitude and longitude (e.g. 64.84, -147.72).");
+        return;
+      }
+
+      const latitude = Number(matches[0]);
+      const longitude = Number(matches[1]);
+      if (latitude > 90 || latitude < -90) {
+        this.setError("Latitude values outside of the range of -90 to 90 may behave unexpectedly.");
+      } else if (longitude > 180 || longitude < -180) {
+        this.setError("Longitude values outside of the range of -180 to 180 may behave unexpectedly.");
+      }
+
+      return { latitude, longitude };
+    },
+
+    clearError() {
+      this.setError("");
+    },
+
+    setError(errorMessage) {
+      this.templateVars.errorMessage = errorMessage;
+      this.render();
+    },
+  });
+
+  return ViewfinderView;
+});
+
+// Regular expression matching a string that contains two numbers optionally separated by a comma.
+const floatsRegex = /[+-]?[0-9]*[.]?[0-9]+/g;
+
+// Regular expression matching everything except numbers, periods, and commas.
+const bannedCharactersRegex = /[^0-9,.+-\s]/g;
+
+// Class names that correspond to elements in the template.
+const classNames = {
+  baseClass: 'viewfinder',
+  button: "viewfinder__button",
+  input: "viewfinder__input",
+};
```

Comment:
I'd like to say yes, unfortunately fields like `event` and `className` cannot reference this.classNames so I don't think it would work. If I instead use a JS class then this should be possible. Can we use `class` in the repo?

Reply:
`events` can also be a method so you can reference `this.classNames` within events like so:

https://github.com/NCEAS/metacatui/blob/main/src/js/views/maps/LayerNavigationView.js#L68-L79

Reply:
Done